src/index/formatFileStructure/findEntryPoints.ts

```diff
@@ -1,53 +1,44 @@
 import { Graph } from "./shared/Graph";
 import { isTestFile } from "./shared/isTestFile";
 
-const invertGraph = (graph: Graph) => {
+function createInvertedGraph(graph: Graph) {
   const invertedGraph: Graph = {};
-  Object.keys(graph).forEach(k => {
-    graph[k].forEach(v => {
-      if (!(v in invertedGraph)) {
-        invertedGraph[v] = [];
-      }
-
-      invertedGraph[v].push(k);
-    });
-  });
-
+  const starts = Object.keys(graph);
+
+  for (const start of starts) {
+    const ends = graph[start];
+    for (const end of ends) {
+      const endMissing = !(end in invertedGraph);
+      if (endMissing) invertedGraph[end] = [];
+      invertedGraph[end].push(start);
+    }
+  }
   return invertedGraph;
-};
+}
 
 export function findEntryPoints(graph: Graph) {
-  const invertedGraph = invertGraph(graph);
+  const invertedGraph = createInvertedGraph(graph);
   const possibleEntryPoints = Object.keys(graph);
-  const entryPoints = possibleEntryPoints.filter(file => {
-    const importedBy = invertedGraph[file];
-    if (!importedBy || !importedBy.length) {
-      return true;
-    }
 
+  const entryPoints = possibleEntryPoints.filter(filePath => {
+    const importedBy = invertedGraph[filePath];
+    if (!importedBy || !importedBy.length) return true;
     return importedBy.every(x => isTestFile(x));
   });
 
-  if (entryPoints.length) {
-    return entryPoints;
-  }
+  if (entryPoints.length) return entryPoints;
 
   const levelMap: Record<number, string[]> = {};
 
-  possibleEntryPoints.forEach(k => {
-    const n = k.split("/").length;
-
-    if (!(n in levelMap)) {
-      levelMap[n] = [];
-    }
-
-    levelMap[n].push(k);
-  });
+  for (const entryPoint of possibleEntryPoints) {
```

Comment:
for of

Reply:
I only prefer using `forEach` when I need the both the value and index for the block of code.

Otherwise, it's faster and neater to use a for each.